src/main/java/io/stargate/sgv2/jsonapi/api/v1/CollectionResource.java

```diff
@@ -183,10 +183,12 @@ else if (error instanceof JsonApiException jsonApiException) {
                   embeddingProvider =
                       embeddingProviderFactory.getConfiguration(
                           dataApiRequestInfo.getTenantId(),
+                          dataApiRequestInfo.getCassandraToken(),
                           vectorizeConfig.provider(),
                           vectorizeConfig.modelName(),
                           collectionProperty.vectorConfig().vectorSize(),
                           vectorizeConfig.parameters(),
+                          vectorizeConfig.authentication(),
                           command.getClass().getSimpleName());
```

Comment:
Since we pass 4 different properties of `vectorizeConfig`, should we just pass `vectorizeConfig` itself and let `getConfiguration()` get whatever it needs from there?